core/base/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -121,6 +126,35 @@ public static String concatPath(Object base, Object... paths) throws IllegalArgu
     return output.toString();
   }
 
+  /**
+   * @param paths the set of paths
+   * @return the lowest common ancestor, or null if paths is null or empty
+   */
+  public static AlluxioURI findLowestCommonAncestor(Collection<AlluxioURI> paths) {
+    if (paths == null || paths.isEmpty()) {
+      return null;
+    }
```

Comment:
Do we know if it is common or rare to have only 1 path in `paths`?

Reply:
Not sure. I thought about about adding a property for a threshold below which we prefer individual listing over recursive (on lca). not sure what a good default would be (maybe 100). 

based on on my test setups, ufs.listStatus times:
- 1 file = 5ms
- recursive list on 1.5k files = 1.5s

The right number for the threshold depends on the number of ufs entries under the sync point / latency to the ufs.

Reply:
what do you think of such as property?